nixos/modules/services/monitoring/staytus.nix

```diff
@@ -0,0 +1,370 @@
+        description = "Set of themes";
+      };
+
+      theme = mkOption {
+        type = types.str;
+        default = "default";
+        description = "The theme to activate";
+      };
+
+      #databaseType = mkOption {
+      #  type = types.str;
+      #  default = "postgresql";
+      #  description = "Type of database";
+      #};
+
+      databaseHost = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = "Database hostname";
+      };
+
+      databasePassword = mkOption {
+        type = types.str;
+        default = "";
+        description = "Database user password";
+      };
+
+      databaseName = mkOption {
+        type = types.str;
+        default = "staytus";
+        description = "Database name";
+      };
+
+      databaseUsername = mkOption {
+        type = types.str;
+        default = "staytus";
+        description = "Database user";
+      };
+
+      smtpHost = mkOption {
+        type = types.str;
+        default = "smtp.deliverhq.com";
+        description = "SMTP hostname";
+      };
+
+      smtpPort = mkOption {
+        type = types.str;
+        default = "25";
+        description = "SMTP port";
+      };
+
+      smtpPassword = mkOption {
+        type = types.str;
+        default = "";
+        description = "SMTP user password";
+      };
+
+      smtpUsername = mkOption {
+        type = types.str;
+        default = "";
+        description = "SMTP user name";
+      };
+
+      secretKeyBase = mkOption {
+        type = types.str;
+        default = "";
+        description = "Secret key base";
+      };
+
+      vhosts = mkOption {
+        type = types.listOf types.str;
+        default = [ "${config.networking.hostName}" ];
+        example = [ "status.yourservice.org" ];
+        description = ''
+          A list of virtual hosts. They must be given as exact names if acme is enabled.
+        '';
+      };
+
+      pumaDebug = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Activate puma debug mode";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      { assertion = cfg.databasePassword != "";
+        message = "services.staytus.databasePassword must be set";
+      }
+    ];
+
+    users.extraUsers = [
+      { name = "staytus";
+        group = "staytus";
+        home = cfg.stateDir;
+      } ];
+
+    users.extraGroups = [
+      { name = "staytus";
+      } ];
+
+    systemd.services.staytusjobs = {
+      after = [ "network.target" "mysql.service" ];
+      wantedBy = [ "multi-user.target" ];
+      environment = staytusEnv;
+
+      path = with pkgs; [
+        config.services.mysql.package
+      ];
+      preStart = ''
+        export HOME=$TMP
+        chown -R staytus:staytus ${cfg.stateDir}
+        chmod -R 755 ${cfg.stateDir}
+
+        rm -rf /run/staytus
+        mkdir -p /run/staytus/pids
+
+        cp -Rv ${package}/share/staytus/* /run/staytus
```

Comment:
How much data is stored here? `/run` is a tmpfs with a limited size, `/var/lib` might be a better choice.

Reply:
Was not aware of that, will change.

Reply:
moved to /var/run/staytus